aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,658 @@
+      <button type="submit">Register</button>
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+  </form>
+
+The data annotations applied to the ``Email`` and ``Password`` properties generate metadata on the model. The Input Tag Helper consumes the model metadata and produces `HTML5 <https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5>`__ ``data-val-*`` attributes (see :doc:`/mvc/models/validation`). These attributes describe the validators to attach to the input fields. This provides unobtrusive HTML5 and `jQuery <https://jquery.com/>`__ validation. The unobtrusive attributes have the format ``data-val-rule="Error Message"``, where rule is the name of the validation rule (such as ``data-val-required``, ``data-val-email``, ``data-val-maxlength``, etc.) If an error message is provided in the attribute, it is displayed as the value for the ``data-val-rule`` attribute. There are also attributes of the form ``data-val-ruleName-argumentName="argumentValue"``, for example, ``data-val-maxlength-max="1024"``.  
+
+HTML Helper alternative to Input Tag Helper
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``Html.TextBox``, ``Html.TextBoxFor``, ``Html.Editor`` and ``Html.EditorFor`` have overlapping features with the Input Tag Helper. The Input Tag Helper will automatically set the ``type`` attribute; ``Html.TextBox`` and ``Html.TextBoxFor`` will not. ``Html.Editor`` and ``Html.EditorFor`` handle collections, complex objects and templates; the Input Tag Helper does not. The Input Tag Helper, ``Html.EditorFor``  and  ``Html.TextBoxFor`` are strongly typed (they use lambda expressions); ``Html.TextBox`` and ``Html.Editor`` are not (they use expression names).
+
+Expression names 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``asp-for`` attribute value is a `ModelExpression <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/ModelExpression/index.html>`__ and the right hand side of a lambda expression. Therefore, ``asp-for="Property1"`` becomes ``m => m.Property1`` in the generated code which is why you don't need to prefix with ``Model``. You can use the "@" character to start an inline expression and move before the ``m.``:
+
+.. code-block:: HTML
+
+  @{
+      var joe = "Joe";
+  }
+  <input asp-for="@joe" />
+
+Generates the following:
+
+.. code-block:: HTML
+  
+    <input type="text" id="joe" name="joe" value="Joe" />
+
+
+Navigating child properties 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can also navigate to child properties using the property path of the view model. Consider a more complex model class that contains a child ``Address`` property.
+
+.. literalinclude::  forms/sample/final/ViewModels/AddressViewModel.cs
+  :language: c#
+  :lines: 5-8
+  :dedent: 3
+  :emphasize-lines: 1-
+
+.. literalinclude::  forms/sample/final/ViewModels/RegisterAddressViewModel.cs
+  :language: c#
+  :lines: 5-11
+  :dedent: 3
+  :emphasize-lines: 6
```

Comment:
My printed document lacks the `Address` property and highlights the `Password` property. Suspect you need `lines: 5-14` and `emphasize-lines: 8`.